test/mobx.js

```diff
@@ -173,39 +192,49 @@ describe('usage with mobx should generally work', () => {
 
     await Tree.add(['root'], {
       key: 'newEmptyFilter',
+      type: 'facet',
       context: {},
     })
     expect(service).to.have.callCount(1)
-    expect(reactor).to.have.callCount(2)
+    expect(reactor).to.have.callCount(7)
     expect(Tree.getNode(['root', 'newEmptyFilter'])).to.exist
 
     await Tree.remove(['root', 'newEmptyFilter'])
     expect(service).to.have.callCount(1)
-    expect(reactor).to.have.callCount(3)
+    expect(reactor).to.have.callCount(8)
     expect(Tree.getNode(['root', 'newEmptyFilter'])).to.not.exist
 
     await Tree.add(['root'], {
       key: 'newFilterWithValueForRemoveTest',
+      type: 'facet',
       data: {
         values: 'asdf',
       },
     })
     expect(service).to.have.callCount(2)
-    expect(reactor).to.have.callCount(5)
+    expect(reactor).to.have.callCount(26)
     expect(Tree.getNode(['root', 'newFilterWithValueForRemoveTest'])).to.exist
 
     await Tree.remove(['root', 'newFilterWithValueForRemoveTest'])
     expect(Tree.getNode(['root', 'newFilterWithValueForRemoveTest'])).to.not
       .exist
     expect(service).to.have.callCount(3)
-    expect(reactor).to.have.callCount(7)
+    expect(reactor).to.have.callCount(42)
 
     expect(
       treeUtils.lookup(['newEmptyFilter'], reactor.getCall(0).args[0])
     ).to.deep.equal({
       key: 'newEmptyFilter',
-      context: {},
+      type: 'facet',
+      lastUpdateTime: null,
       path: ['root', 'newEmptyFilter'],
+      error: null,
```

Comment:
instead of looking at all these nulls, maybe we could use something less strict than deep equal?